internal/pipe/sign/sign.go

```diff
@@ -81,14 +81,21 @@ func (Pipe) Run(ctx *context.Context) error {
 					artifact.ByType(artifact.Checksum),
 					artifact.ByType(artifact.LinuxPackage),
 				))
-				if len(cfg.IDs) > 0 {
-					filters = append(filters, artifact.ByIDs(cfg.IDs...))
-				}
+			case "archive":
+				filters = append(filters, artifact.ByType(artifact.UploadableArchive))
+			case "binary":
+				filters = append(filters, artifact.ByType(artifact.UploadableBinary))
+			case "package":
+				filters = append(filters, artifact.ByType(artifact.LinuxPackage))
 			case "none":
 				return pipe.ErrSkipSignEnabled
 			default:
 				return fmt.Errorf("invalid list of artifacts to sign: %s", cfg.Artifacts)
 			}
+
+			if cfg.Artifacts != "checksum" && len(cfg.IDs) > 0 {
+				filters = append(filters, artifact.ByIDs(cfg.IDs...))
+			}
```

Comment:
Just to clarify - do we not do the filtering by both IDs and checksum because there will always be only one checksum for the release? If so, then I'm assuming this then also applies to `source` - which is not part of the condition here?

Reply:
the id filter skips itself when the type is checksum

https://github.com/goreleaser/goreleaser/blob/master/internal/artifact/artifact.go#L227-L230

I think this if there was from before this... or just to be extra sure... not really remember 🤔

Reply:
I don't even need the if there actually, forgot it there 🤣